### DISCUSSION TITLE
How does mp4 write work?

### DISCUSSION THREAD

**Discussion by Clauber** [gtxaspec/wz_mini_hacks#336] [Q&A] [answered]
Please, forgive my noobness. I did search in the discussions for this and couldn't find an answer.
I just barely installed wz_mini_hacks on my cam v3 and I'm loving it.
I do have a samba server at home and a couple cloud providers with a lot of storage, and would love to store my footage in there.
I noticed that wz_mini_hacks has mp4_write option, which, according to the documentation, allows me to write to my storage medium.

How do I set up so the camera knows where to store my videos? After changing that setting to true, what else do I have to do?

Thanks a lot.

**Comment by kohrar** [comment, marked as answer]
As far as I'm aware, the `ENABLE_MP4_WRITE` option only enables mp4 recordings to the SD card only (in `/media/mmc/record/...`).

If you want the recordings to go directly to your samba server, you could try enabling this option and then mounting your samba/cifs share over this path either manually (Eg. `mount -t cifs -o username=x,password=y,vers=2.1 //$smb-server/$surveillance-share /media/mmc/record`) or with a fstab entry (eg. `//$smb-server/$surveillance-share      /media/mmc/record     cifs   username=x,password=y,vers=2.1  0 0`).

**Reply by Clauber** [reply]
Looks like I completely misunderstood what that option is doing.

Thank you so much for clarifying and also for proposing a solution. 
I'm a software engineer, not a sys admin, so I had no idea it was possible to mount a remote folder locally. I just tried mounting the path manually and it worked. 

I'm also aware it's possible to create custom startup scripts. I'm gonna go ahead and read the documentation on that and make sure I get it working automatically. 

Thank you so much!

**Reply by mrhollywoodgates** [reply]
I tried to mount a windows SMB share without luck, using the command above. I would get "invalid option" error. Removing the `vers` option changed errors to "host unreachable", which led me to enable smbv1 on my Windows server. Now I get "permission denied". 
My full command I'm using is 
```
mount -t cifs -o username=x,password=y //$smbIP/$share /media/mmc/record
```
the /media/mmc/record is the mounted SD card where the camera wants to record things, and I can create files via the shell just fine. 
Any advice?

**Reply by iansmith42** [reply]
I have put both NFS and CIFS entries into fstab and can mount them manually via ssh, but when I reboot the camera they are not being mounted.

Do I need to add a startup script to manually run "mount /media/mmc/record" or is there something I am missing?

**Comment by virmaior** [comment]
For reference I actually do the opposite.
I run a cron job on my raspberry pi to download the videos of the previous hour and prepare them for processing. This actually gives me a backup in case something screws up on the pi.

**Reply by baudev** [reply]
Yeah, but continuously recording a video stream on an SD card may damage it 
I did it for 6 months on 2 wyze cam and both cards died...

**Comment by gtxaspec** [comment]
Heres what the mp4write function does:

Normally, the camera's factory firmware does this when recording:

1. record video file to `/tmp/video.mp4'
2. move the video file from `/tmp/video.mp4` to `/sdcard/record/date/xx.mp4`

the process of moving the video file from `/tmp` to `/sdcard` can cause whats called trashing, causing the camera to lag or crash.  See https://en.wikipedia.org/wiki/Thrashing_(computer_science)

What mp4write does, is this:

1. record video directly to  `/sdcard/record/date/xx.mp4`

with this function, we avoid any issues with camera performance.  The latest wyze firmware has integrated this fix (don't upgrade if you want rtsp)

hope this clears it up.

**Reply by clee604** [reply]
does this still respect the option in wyze for when it is recording? ie. on motion detection only? Or does mp4write function enable continuous writing?

if it follows the same recording conditions as the wyze app, are there any reasons why we shoudln't enable mp4write if it improves performance and reduces thrashing?

**Reply by gtxaspec** [reply]
it functions on any and all writes to the sdcard, i generally enable it on all cameras.

**Reply by clee604** [reply]
thanks!

**Reply by clee604** [reply]
completely unrelated but quick question, do you do anything special before unplugging the camera (to say move it or take the sd card out) to prevent corruption? I tried a shutdown now command but it doesn't work.. the other thing i could think of was use the wyze camera to eject the sd card
